ionoscloud/resource_server.go

```diff
@@ -1001,15 +1015,15 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 			mProp, _ := json.Marshal(nicProperties)
 			log.Printf("[DEBUG] Updating props: %s", string(mProp))
 			var createdNic ionoscloud.Nic
-			if !updateNic {
+			if createNic {
 				nic.Properties = &nicProperties
 				createdNic, apiResponse, err = client.NetworkInterfacesApi.DatacentersServersNicsPost(ctx, d.Get("datacenter_id").(string), *server.Id).Nic(*nic).Execute()
-			} else {
+			} else if nic.Id != nil {
```

Comment:
@cristiGuranIonos is this check really needed? not sure if there will be a case in which we want to update a `nic` and the `nic` has no ID.

Reply:
just to be sure, but if createNic is set wrong in some corner case, I don't want it to crash